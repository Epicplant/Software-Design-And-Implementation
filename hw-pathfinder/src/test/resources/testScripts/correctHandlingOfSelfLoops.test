# Tests that if there are multiple self loops littered throughout the graph that the minimum cost path
# is still calculated correctly.

# Creates a new graph
CreateGraph newGraph

# Adds node A to the graph
AddNode newGraph A

# Adds node B to the graph
AddNode newGraph B

# Adds node C to the graph
AddNode newGraph C

# Adds node D to the graph
AddNode newGraph D

# Adds edge of weight 2.5 to the graph
AddEdge newGraph A A 2.500

# Adds edge of weight 3 to the graph
AddEdge newGraph A B 3

# Adds edge of weight 1 to the graph
AddEdge newGraph B B 1

# Adds edge of weight 3 to the graph
AddEdge newGraph B C 3

# Adds edge of weight 4 to the graph
AddEdge newGraph C C 4

# Adds edge of weight 2.8 to the graph
AddEdge newGraph C D 2.800

# Adds edge of weight 1 to the graph
AddEdge newGraph D D 1

# Finds the shortest path from node A to node D
FindPath newGraph A D