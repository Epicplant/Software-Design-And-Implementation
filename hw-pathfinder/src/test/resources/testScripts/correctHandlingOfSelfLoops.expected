# Tests that if there are multiple self loops littered throughout the graph that the minimum cost path
# is still calculated correctly.

# Creates a new graph
created graph newGraph

# Adds node A to the graph
added node A to newGraph

# Adds node B to the graph
added node B to newGraph

# Adds node C to the graph
added node C to newGraph

# Adds node D to the graph
added node D to newGraph

# Adds edge of weight 2.5 to the graph
added edge 2.500 from A to A in newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from A to B in newGraph

# Adds edge of weight 1 to the graph
added edge 1.000 from B to B in newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from B to C in newGraph

# Adds edge of weight 4 to the graph
added edge 4.000 from C to C in newGraph

# Adds edge of weight 2.8 to the graph
added edge 2.800 from C to D in newGraph

# Adds edge of weight 1 to the graph
added edge 1.000 from D to D in newGraph

# Finds the shortest path from node A to node D
path from A to D:
A to B with weight 3.000
B to C with weight 3.000
C to D with weight 2.800
total cost: 8.800
